File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
import java.util.*;

class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>(nums.length * 2);

        for (int num : nums) {
            set.add(num);
        }

        int longest = 0;

        for (int num : set) {
            if (!set.contains(num - 1)) {
                int curr = num;

                while (set.contains(curr)) {
                    curr++;
                }

                longest = Math.max(longest, curr - num);
            }
        }

        return longest;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums):
        num_set = set(nums)
        longest = 0

        for num in num_set:
            if num - 1 not in num_set:  
                curr = num

                while curr in num_set:
                    curr += 1

                longest = max(longest, curr - num)

        return longest